ras/uprobe.py

```diff
@@ -36,27 +35,35 @@ def run_cmd(self, cmd):
 
     @staticmethod
     def run_cmd_out(cmd):
-        return process.system_output(cmd, shell=True, ignore_status=True,
+        return process.system_output(cmd,
+                                     shell=True, ignore_status=True,
                                      sudo=True).decode("utf-8")
 
     def clear_trace(self):
-        self.debugfs = "/sys/kernel/debug/tracing/"
-        self.tracefs = os.path.join(self.debugfs, "trace")
-        self.uprobes_enable_fs = os.path.join(self.debugfs, "events/uprobes/enable")
-        self.uprobes_event_fs = os.path.join(self.debugfs, "events/uprobes/u_malloc/enable")
-        self.uprobes_events_fs = os.path.join(self.debugfs, "uprobe_events")
-        genio.write_one_line(self.tracefs, "")
-        genio.write_one_line(self.uprobes_events_fs, "")
-
-    def enable_uprobes(self):
-        genio.write_one_line(self.uprobes_enable_fs, "1")
-        genio.write_one_line(self.uprobes_event_fs, "1")
+        try:
+            self.debugfs = "/sys/kernel/debug/tracing/"
+            self.tracefs = os.path.join(self.debugfs, "trace")
+            uprobes_en = "events/uprobes/enable"
+            self.uprobes_enable_fs = os.path.join(self.debugfs, uprobes_en)
+            uprobes_umall = "events/uprobes/u_malloc/enable"
+            self.uprobes_event_fs = os.path.join(self.debugfs, uprobes_umall)
+            uprobe_eve = "uprobe_events"
+            self.uprobes_events_fs = os.path.join(self.debugfs, uprobe_eve)
+            genio.write_one_line(self.tracefs, "")
+            genio.write_one_line(self.uprobes_events_fs, "")
+        except IOError:
+            self.cancel("There is issue with kernel system resources")
 
-    def disable_uprobes(self):
-        genio.write_one_line(self.uprobes_event_fs, "0")
-        genio.write_one_line(self.uprobes_enable_fs, "0")
+    def ena_dis_uprobes(self, prob_value):
```

Comment:
@SamirMulani can you please add docstrings to all the methods.
I don't see to any of the method. having doctrings methods are more self explanatory :-)